From castet.matthieu@free.fr Tue Oct  3 12:46:45 2006
Message-ID: <4522BE19.1070103@free.fr>
Date: Tue, 03 Oct 2006 21:46:33 +0200
From: matthieu castet <castet.matthieu@free.fr>
To: greg@kroah.com
CC: linux-kernel@vger.kernel.org,  usbatm@lists.infradead.org, linux-usb-devel@lists.sourceforge.net, ueagle <ueagleatm-dev@gna.org>
Subject: [PATCH 1/3] UEAGLE : be suspend friendly

this patch avoid that the kernel thread block the suspend process.
Some work is still need to recover after a resume.

Signed-off-by: Matthieu Castet <castet.matthieu@free.fr>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/atm/ueagle-atm.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/usb/atm/ueagle-atm.c
+++ gregkh-2.6/drivers/usb/atm/ueagle-atm.c
@@ -1173,6 +1173,9 @@ static int uea_kthread(void *data)
 			ret = uea_stat(sc);
 		if (ret != -EAGAIN)
 			msleep(1000);
+ 		if (try_to_freeze())
+			uea_err(INS_TO_USBDEV(sc), "suspend/resume not supported, "
+				"please unplug/replug your modem\n");
 	}
 	uea_leaves(INS_TO_USBDEV(sc));
 	return ret;
